src/fhir-products/fhir-products.controller.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  Controller,
+  Get,
+  Query,
+  BadGatewayException,
+  NotFoundException,
+  Param,
+} from '@nestjs/common';
+import { R4 } from '@ahryman40k/ts-fhir-types';
+import { ProductsService } from 'src/products/products.service';
+import { ProductsQuery } from 'src/common/interfaces/products-query.interface';
+import {
+  formatProductsToFhir,
+  getSystems,
+} from 'src/common/utils/fhir/formatter';
+import { ProductDto } from 'src/common/dtos/product.dto';
+
+@Controller('fhir')
+export class FhirProductsController {
+  constructor(private service: ProductsService) {}
+
+  @Get()
+  async getFhir(@Query() query: ProductsQuery): Promise<R4.IConceptMap> {
```

Comment:
Since the root is supposed to get >= 1 concept maps then perhaps returning a Bundle resource may be more appropriate. Not entirely sure.

Reply:
I also thought so in the first place but then from what I understood, the conceptMap is an object that contains all mappings of 1 system to the targeted systems.

Reply:
Interesting, okay. May you share more resources on that?